src/objects/application.ts

```diff
@@ -32,4 +37,12 @@ function createApplication(stage: HTMLDivElement): void {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     (globalThis as any).__PIXI_APP__ = application
   }
+}
+
+export async function waitForApplication(): Promise<Application> {
+  if (application) {
+    return await application
+  }
+
+  return waitForEvent('applicationCreated')
```

Comment:
It might be worth creating enums for these so we can reference like `event.ApplicationCreated` - depending on how much we plan to use them

Reply:
Personally I think the typesafe strings are preferred. All the same typesafety without the additional import.

Reply:
fair enough 🤷🏻